stix2/base.py

```diff
@@ -125,6 +128,11 @@ def _check_at_least_one_property(self, list_of_properties=None):
                 list_of_properties.remove('type')
         current_properties = self.properties_populated()
         list_of_properties_populated = set(list_of_properties).intersection(current_properties)
+
+        if list_of_properties_populated == set(['id']) and isinstance(self, _Observable):
+            # Do not count the auto-generated id as a user-specified property
+            list_of_properties_populated = None
+
```

Comment:
This implies that if "id" was included in the property group (i.e. that which is subject to the "at least one" presence co-constraint), and that property is the only one present (in an observable), produce an error.  I don't understand the reasoning.  Why include "id" in the group at all?

For that matter, the same goes for the "extensions" property for all objects.  Why would you include that in the group then?

Seems like this ought to be a basic presence co-constraint check, but there's weird special casing going on that's hard to understand.  More comments/docstring would be nice...